sqldelight-compiler/src/main/kotlin/app/cash/sqldelight/core/compiler/SqlDelightCompiler.kt

```diff
@@ -42,12 +42,13 @@ object SqlDelightCompiler {
     module: Module,
     dialect: SqlDelightDialect,
     file: SqlDelightQueriesFile,
-    output: FileAppender
+    output: FileAppender,
+    generateAsync: Boolean,
```

Comment:
instead of passing this around I would just pull it from the dialect, similar to [this](https://github.com/cashapp/sqldelight/blob/master/sqldelight-compiler/src/main/kotlin/app/cash/sqldelight/core/compiler/QueryGenerator.kt#L37)

Reply:
This will also fix the currently failing intellij tests, and probably clean up some code